File: org.summer.view.widget/src/org/summer/view/widget/model/IBindingList.java
package org.summer.view.widget.model;

import org.summer.view.widget.collection.ICollection;
import org.summer.view.widget.collection.IEnumerable;
import org.summer.view.widget.collection.IList;

/// <summary>
	/// Provides the features required to support both complex and simple scenarios when binding to a data source.
	/// </summary>
	public interface IBindingList extends IList, ICollection, IEnumerable
	{
		void AddIndex (PropertyDescriptor property);

		Object AddNew ();

		void ApplySort (PropertyDescriptor property, ListSortDirection direction);

		int Find (PropertyDescriptor property, Object key);

		void RemoveIndex (PropertyDescriptor property);

		void RemoveSort ();

		boolean AllowEdit {
			get;
		}

		boolean AllowNew {
			get;
		}

		boolean AllowRemove {
			get;
		}

		boolean IsSorted {
			get;
		}

		ListSortDirection SortDirection {
			get;
		}

		PropertyDescriptor SortProperty {
			get;
		}

		boolean SupportsChangeNotification {
			get;
		}

		boolean SupportsSearching {
			get;
		}

		boolean SupportsSorting {
			get;
		}

		/*event*/ ListChangedEventHandler ListChanged;
	}

File: summer/WebContent/js/system/componentmodel/IBindingList.js
/**
 * IBindingList
 */

define(["dojo/_base/declare", "system/Type",
        "collections/IList", "collections/ICollection", "collections/IEnumerable"], 
		function(declare, Type,
				IList, ICollection, IEnumerable){
	var IBindingList = declare("IBindingList", [IList, ICollection, IEnumerable],{
	});
	
	IBindingList.Type = new Type("IBindingList", IBindingList, [IList.Type, ICollection.Type, IEnumerable.Type]);
	return IBindingList;
});
//
//public interface IBindingList : IList, ICollection, IEnumerable
//	{
//		event ListChangedEventHandler ListChanged;
//		bool AllowNew
//		{
//			get;
//		}
//		bool AllowEdit
//		{
//			get;
//		}
//		bool AllowRemove
//		{
//			get;
//		}
//		bool SupportsChangeNotification
//		{
//			get;
//		}
//		bool SupportsSearching
//		{
//			get;
//		}
//		bool SupportsSorting
//		{
//			get;
//		}
//		bool IsSorted
//		{
//			get;
//		}
//		PropertyDescriptor SortProperty
//		{
//			get;
//		}
//		ListSortDirection SortDirection
//		{
//			get;
//		}
//		object AddNew();
//		void AddIndex(PropertyDescriptor property);
//		void ApplySort(PropertyDescriptor property, ListSortDirection direction);
//		int Find(PropertyDescriptor property, object key);
//		void RemoveIndex(PropertyDescriptor property);
//		void RemoveSort();
//	}